src/popup/utils/utils.ts

```diff
@@ -33,28 +41,56 @@ export function formatTime(time: number) {
   });
 }
 
-export function checkHashType(fullHash: string) {
+export function getHashPrefix(fullHash: string) {
```

Comment:
I think this function should not be used within the `validateHash` but the `validateHash` should be used here. So the opposite.